src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -78,9 +84,39 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
 
             var compilation = this.DeclaringCompilation;
 
-            // Dev11 doesn't synthesize this attribute, the debugger has a knowledge 
+            // do not emit CompilerGenerated attributes for fields inside compiler generated types:
+            if (!this.ContainingType.IsImplicitlyDeclared)
```

Comment:
>if (!this.ContainingType.IsImplicitlyDeclared) [](start = 12, length = 46)

I do not think this condition can ever be true. Consider replacing the ```if``` with assert.

Reply:
It turns out this can be reached (`InteractiveSessionTests.ScriptMemberAccessFromNestedClass`). I'll restore the original code.